api/src/main/java/org/apache/iceberg/actions/DeleteOrphanFiles.java

```diff
@@ -85,8 +85,33 @@ public interface DeleteOrphanFiles extends Action<DeleteOrphanFiles, DeleteOrpha
    */
   interface Result {
     /**
+     * @deprecated since 0.14.0, will be removed in 0.15.0; use {@link #orphanFiles()} instead.
      * Returns locations of orphan files.
      */
+    @Deprecated
     Iterable<String> orphanFileLocations();
+
+    /**
+     * Returns orphan files.
+     */
+    Iterable<OrphanFileStatus> orphanFiles();
```

Comment:
I wonder if it makes sense to have a method like, failures() that is dedicated to returning failure statuses.  The status isnt very helpful if the delete was a success